# Zillow Clustering Project Data Acquisition Notebook

In [1]:
# imports
import pandas as pd
import numpy as np
import acquire
import warnings
warnings.filterwarnings('ignore')

In [2]:
# connect to database, perform SQL query, write to csv, and convert to pandas DataFrame
df = acquire.get_zillow_cluster()

In [4]:
# what do we have to start with?
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 52438 entries, 0 to 52437
Data columns (total 68 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   typeconstructiontypeid        76 non-null     float64
 1   storytypeid                   47 non-null     float64
 2   propertylandusetypeid         52438 non-null  float64
 3   heatingorsystemtypeid         33934 non-null  float64
 4   buildingclasstypeid           0 non-null      object 
 5   architecturalstyletypeid      70 non-null     float64
 6   airconditioningtypeid         13638 non-null  float64
 7   parcelid                      52438 non-null  int64  
 8   id                            52438 non-null  int64  
 9   basementsqft                  47 non-null     float64
 10  bathroomcnt                   52438 non-null  float64
 11  bedroomcnt                    52438 non-null  float64
 12  buildingqualitytypeid         33739 non-null  float64
 13  c

### Data Aquistion Key Findings & Takeaways:
- This dataframe has 52438 rows and 68 columns.
- Target variable: 'logerror'
- There are 14 columns that are dtype "object" that either need to be dropped, or converted to int
- There are numerous columns with mostly missing values that either need to be dropped, or imputed
- There are numerous columns that at first glance appear to be duplicates, or similar enough to either be dropped, or joined
- Create Columns with county names
- Create Variables:
    - age
    - acres
    - bath:bed ratio
    - taxrate

#### Data Preparation To Do List:
- Set parcelid as index
- Encode or Drop Object Columns
- Create New Variable Columns for:
    - age of property
    - convert lot size to acres
    - bath:bed ratio
    - taxrate
- Drop Columns that have more than 50% null values
- Drop Rows that have more than 75% null values
- Other Columns to Drop:
    - propertylandusetypeid
    - id
    - finishedsqft12
    - assessmentyear